OneSignalSDK/build.gradle

```diff
@@ -55,4 +59,22 @@ allprojects {
         // Huawei maven
         maven { url 'https://developer.huawei.com/repo/' }
     }
+    
+    // Force all modules to use the same Kotlin version
+    configurations.all {
+        resolutionStrategy {
+            force "org.jetbrains.kotlin:kotlin-stdlib:$kotlinVersion"
+            force "org.jetbrains.kotlin:kotlin-stdlib-common:$kotlinVersion"
+            
+            // Exclude deprecated jdk7/jdk8 variants
+            eachDependency { details ->
+                if (details.requested.group == 'org.jetbrains.kotlin') {
+                    if (details.requested.name == 'kotlin-stdlib-jdk7' || 
+                        details.requested.name == 'kotlin-stdlib-jdk8') {
+                        details.useTarget "org.jetbrains.kotlin:kotlin-stdlib:$kotlinVersion"
+                    }
+                }
+            }
+        }
+    }
```

Comment:
Recommend we make sure this substitution works before publishing the 5.4.0 release. I have concerns if this really works to correctly when it comes to the dependencies in the .aar files.